src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,122 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.time.LocalDate;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import task.*;
+import duke_exception.*;
+
+/**
+ * Storage handles loading/saving of task list data from/to file.
+ *
+ * @author Ho Wen Zhong
+ */
+public class Storage {
+
+    private String filePath;
+    private File file;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        this.file = new File(filePath);
+    }
+
+    /**
+     * Loads the task list data from file.
+     *
+     * @return List of Tasks.
+     * @throws DukeException If there are file access errors.
+     */
+    public List<Task> load() throws DukeException {
```

Comment:
Did not catch DateTimeException that can be thrown by LocalDate.parse()

```suggestion
    public List<Task> load() throws DukeException, DateTimeException {
```